File: javascript/util/combination.js
function getCombos(arr, pick) {
  if (!pick) {
    return [[]];
  }

  if (!arr.length) {
    return [];
  }

  const first = arr[0];
  const rest = arr.slice(1);

  return getCombos(rest, pick - 1)
    .map((combo) => [first].concat(combo))
    .concat(getCombos(rest, pick));
}

module.exports = { getCombos };

File: python/util/combination.py
def get_combos(arr, pick):
  if pick == 0:
    return [[]]

  if len(arr) < 1:
    return []

  first = arr[0]
  rest = arr[1:]

  next_combos = get_combos(rest, pick - 1)
  # essentially map()
  for combo in next_combos:
    combo.insert(0, first)

  return next_combos + get_combos(rest, pick)